plugins/destination/duckdb/client/client.go

```diff
@@ -89,6 +78,29 @@ func (c *Client) Close(ctx context.Context) error {
 	return err
 }
 
+func TestConnection(ctx context.Context, _ zerolog.Logger, specBytes []byte) error {
+	var s Spec
+	if err := json.Unmarshal(specBytes, &s); err != nil {
+		return &plugin.TestConnError{
+			Code:    plugin.TestConnFailureCodeInvalidSpec,
+			Message: fmt.Errorf("failed to unmarshal spec: %w", err),
+		}
+	}
+	s.SetDefaults()
+
+	connector, err := duckdb.NewConnector(amendConnectionString(s.ConnectionString), nil)
+	if err != nil {
+		return err
```

Comment:
go-duckdb errors seem to be clear and not so deeply wrapped, that's why I omitted the `processError(err)` step here.